Source/Core/VideoCommon/Assets/WatchableFilesystemAssetLibrary.h

```diff
@@ -0,0 +1,38 @@
+// Copyright 2024 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include <map>
+#include <memory>
+#include <string>
+#include <string_view>
+
+#include <wtr/watcher.hpp>
+
+#include "VideoCommon/Assets/CustomAssetLibrary.h"
+
+namespace VideoCommon
+{
+class WatchableFilesystemAssetLibrary : public CustomAssetLibrary
```

Comment:
Would it make sense for this filesystem watcher to be less "Asset" oriented and be renamed `FilesystemWatcher` or whatever? We use Qts equivalent for the game list, that could be changed, but there's probably other places it could be used. e.g. "Pipes" input backend, "GCI Folder" maybe?

Reply:
Sure, I'd be happy to move this interface to somewhere else if it makes more sense.  I do need to retain `WatchableFilesystemAssetLibrary` because it is used both by `DirectFilesystemAssetLibrary` and the editor library.  But it could be something like this:

```
class WatchableFilesystemAssetLibrary : public CustomAssetLibrary, Common::FilesystemWatcher
{};
```

Do you have a preference of where and what it is called?

Reply:
`FilesystemWatcher` in `Common` sounds fine, yeah?